android/app/src/main/java/com/tourneynizer/tourneynizer/activities/TournamentListActivity.java

```diff
@@ -0,0 +1,100 @@
+package com.tourneynizer.tourneynizer.activities;
+
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.location.Address;
+import android.location.Geocoder;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.Gravity;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.LinearLayout;
+import android.widget.ListView;
+import android.widget.ProgressBar;
+
+import com.tourneynizer.tourneynizer.R;
+import com.tourneynizer.tourneynizer.adapters.TournamentListAdapter;
+import com.tourneynizer.tourneynizer.data.Tournament;
+import com.tourneynizer.tourneynizer.data.TournamentType;
+
+import java.io.IOException;
+import java.sql.Time;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+
+public class TournamentListActivity extends AppCompatActivity {
+
+    public static final String INTENT_TOURNAMENT = "com.dreamteam.tournament.data.Tournament";
+    private TournamentListAdapter listAdapter;
+
+    // will delete these members when back end is exposed please disregard
+    private Geocoder geocoder;
+    private List<Address> a1;
+    private List<Address> a2;
+    private Bitmap logo;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```

Comment:
This function looks good, but in the future try to utilize helper functions.

Reply:
Thanks I will